website/client/components/tasks/spells.vue

```diff
@@ -236,35 +237,39 @@ export default {
       this.spell = spell;
 
       if (spell.target === 'self') {
-        this.castEnd(null, 'self');
+        this.castEnd(null, spell.target);
       } else if (spell.target === 'party') {
         if (!this.user.party._id) {
           let party = [this.user];
-          this.castEnd(party, 'party');
+          this.castEnd(party, spell.target);
           return;
         }
-
-        // @TODO: do we need to fetcht the party everytime? We should probably just check store
+        // @TODO: do we need to fetch the party everytime? We should probably just check store. But what if a new member has arrived recently?
         let party = await this.$store.dispatch('guilds:getGroup', {groupId: 'party'});
         party = isArray(party) ? party : [];
         party = party.concat(this.user);
         this.$store.state.party.data = party;
-        this.castEnd(party, 'party');
-      } else if (spell.target === 'tasks') {
+        this.castEnd(party, spell.target);
+      } else if (spell.target === 'tasks' || spell.target === 'task') {
+        // cast on all 'tasks' or one 'task' (Rewards are not allowed)
```

Comment:
You definitely need that. Skills that act on one task have a target of `task`, skills that act on all, like Searing Brightness, have a target of `tasks`. Search for "brightness" and "fireball" in website/common/script/content/spells.js for examples. I suggest including the comment to make that clearer for future programmers too, but feel free to improve the comment because it's clearly not clear enough. :)